internal/services/netapp/netapp_volume_group_resource.go

```diff
@@ -0,0 +1,642 @@
+package netapp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumegroups"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumes"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumesreplication"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	netAppValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NetAppVolumeGroupResource struct{}
+
+type NetAppVolumeGroupModel struct {
+	Name                  string                    `tfschema:"name"`
+	ResourceGroupName     string                    `tfschema:"resource_group_name"`
+	Location              string                    `tfschema:"location"`
+	AccountName           string                    `tfschema:"account_name"`
+	GroupDescription      string                    `tfschema:"group_description"`
+	ApplicationType       string                    `tfschema:"application_type"`
+	ApplicationIdentifier string                    `tfschema:"application_identifier"`
+	Volumes               []NetAppVolumeGroupVolume `tfschema:"volume"`
+}
+
+var _ sdk.Resource = NetAppVolumeGroupResource{}
+
+func (r NetAppVolumeGroupResource) ModelObject() interface{} {
+	return &NetAppVolumeGroupModel{}
+}
+
+func (r NetAppVolumeGroupResource) ResourceType() string {
+	return "azurerm_netapp_volume_group"
+}
+
+func (r NetAppVolumeGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumegroups.ValidateVolumeGroupID
+}
+
+func (r NetAppVolumeGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.AccountName,
+		},
+
+		"group_description": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"application_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(volumegroups.ApplicationTypeSAPNegativeHANA),
```

Comment:
Yes, there are plans to add more and the second is coming up in the next 2-4 months.

Reply:
I really don't see a reason for that, this is not a new type of volume, it will be diverging from our main documentation, we are not creating a new type of volume, the volume spec is only used to indicate the backend how to handle volume creation and what rules to use for validating it and how to best place the volume within our datacenters.

It will cause a lot of confusion for our customers.

Reply:
Based on our offline discussion over Slack, I'll be working on this change. Thanks for the good points.

Reply:
Just pushed the changes and here an acceptance test successfully executing:

```
[ ~/go/src/github.com/terraform-providers/terraform-provider-azurerm ]$ make acctests SERVICE='netapp' TESTARGS=' -parallel 1 -run=TestAccNetAppVolumeGroupSapHana_basic' TESTTIMEOUT='600m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/netapp -parallel 1 -run=TestAccNetAppVolumeGroupSapHana_basic -timeout 600m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNetAppVolumeGroupSapHana_basic
=== PAUSE TestAccNetAppVolumeGroupSapHana_basic
=== CONT  TestAccNetAppVolumeGroupSapHana_basic
--- PASS: TestAccNetAppVolumeGroupSapHana_basic (1704.00s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp        1704.031s
```